Src/BootCamp.Chapter/Inventory.cs

```diff
@@ -29,7 +93,37 @@ public void AddItem(Item item)
         /// </summary>
         public void RemoveItem(Item item)
         {
+            _items = RemoveFromArray(item);
+        }
+
+        public Item[] RemoveFromArray(Item item)
+        {
+            if (_items.Length == 0)
+            {
+                return _items;
+            }
+
+            var newArray = new Item[_items.Length - 1];
+            var found = false;
+            for (int i = 0; i < _items.Length - 1; i++)
```

Comment:
i was looking this over and wondering. if you have an array of 
string[2] { "Sword"[0], "Shield"[1]}
i=0... _items.Length -1 = 1
 Item item = sword. 
so it goes through the array once finds sword so found = true. Then tries to go for a second run because of continue  but does not because i = 1 and _items.Length = 1... so the newArray will be empty ?

Reply:
no.

I hope I say it well. 

Let's say we have  a inventory of 2 items.  {sword, shield} 
and we want to remove the sword. 

and we have this code : 
``` 
var newArray = new Item[_items.Length - 1];
            var found = false;
            for (int i = 0; i < _items.Length - 1; i++)
            {
                if (_items[i] == item)
                {
                    found = true;
                    continue;
                }

                if (found)
                {
                    newArray[i - 1] = _items[i];
                }
                else
                {
                    newArray[i] = _items[i];
                }
            }
```  
so new array will be a array of 1 (  2 - 1 = 1) 
found = false 
so the loop begins so i = 0 
then item[i] = "sword"   which is equal to what we want to delete. 
so found = true. 
and we continue witht the loop without the rest of the code. Continue does that. 
So i will be 1 
item[1] = shield.
that is not equal to sword so we goto the rest of the code 
then I check if the item is found, which it is. 
so we goto this line : 
newArray[i - 1] = _items[i];
which will be  array[1-1] = items[i] 
so array[0]  will be shield because that one is on position 1 of the items array. 
then i will be 2 which is not smaller then 2 - 1 = 1 so the loop stops.

so newArray will be a array with 1 item the shield. 

I hope I explain it well.